app/controllers/requests.rb

```diff
@@ -0,0 +1,89 @@
+class MakersBnb < Sinatra::Base
+
+  get '/requests' do
+    if current_user
+      @users_requests = Request.all(user_id: current_user.id)
+      @received_requests = Request.all(owner_id: current_user.id)
+      @request_display = @users_requests.map do |x|
+        Space.get(x.space_id)
+      end
+      erb :'requests/index'
+    else
+      redirect to '/sessions/new'
+    end
+  end
+
+  post '/requests/new' do
+
+    requested_date = Date.parse(params[:date_requested]) rescue nil
+
+    unless requested_date
+      flash[:notice] = 'Please enter a valid date'
+      redirect '/spaces'
+    end
+
+    unless session[:available_dates].include?(Date.parse(params[:date_requested]).strftime)
```

Comment:
Why are you checking the `session` for `available_dates`? Shouldn't the model know about this?